src/pages/ServiceDetails/ServiceMetrics.tsx

```diff
@@ -49,6 +52,9 @@ type NamedHistogram = {
 };
 
 class ServiceMetrics extends React.Component<ServiceId, ServiceMetricsState> {
+  static contextTypes = {
```

Comment:
What is it for?

Reply:
You can't access to the store of the provider if you not define store in the contextTypes, you can't access with this.context . 
React only populates the context if you specify contextTypes in your component. You will however notice that the props in your component is populated from the store, as you specified in your mapStateToProps function. If you wanna access them without use contextTypes you must handle the subscriptions,... on the root level (where you create the store)

Reply:
You really shouldn't need to access the store you get whatever state you want from the store just by specifying the props in `mapStateToProps` and using `connect`.